requests/auth.py

```diff
@@ -166,12 +167,12 @@ def handle_401(self, r, **kwargs):
             # Rewind the file position indicator of the body to where
             # it was to resend the request.
             r.request.body.seek(self.pos)
-        num_401_calls = getattr(self, 'num_401_calls', 1)
+        num_401_calls = getattr(self.num_401_calls, 'value', 1)
```

Comment:
I just adapted the existing code and as such left the `getattr` as it was. After inspection it doesn't seem needed, however: no fallback to default 1 will ever occur because the (now thread-local) value is assigned in __call__, when the authentication process is about to "begin".

Additionally, it shouldn't be needed in the code before this commit: `self.num_401_calls` was being assigned in `__init__`.

Would you like me to update the PR with an extra commit?